src/libbson/src/bson/bson-string.c

```diff
@@ -116,7 +116,44 @@ bson_string_new (const char *str) /* IN */
 
    ret->str[len_u32] = '\0';
    ret->len = len_u32;
+   return ret;
+}
+
+/*
+ *--------------------------------------------------------------------------
+ *
+ * bson_string_alloc --
+ *
+ *       Create an empty bson_string_t and allocate memory for it.
+ *
+ *       The amount of memory allocated will be the next power-of-two if the
+ *       specified size is not already a power-of-two.
+ *
+ * Parameters:
+ *       @size: Size of the string to allocate
+ *
+ * Returns:
+ *       A newly allocated bson_string_t that should be freed with
+ *       bson_string_free().
+ *
+ * Side effects:
+ *       None.
+ *
+ *--------------------------------------------------------------------------
+ */
+
+bson_string_t *
+bson_string_alloc (const uint32_t size)
```

Comment:
```suggestion
bson_string_alloc (const size_t size)
```
Use `size_t` in the parameter (as done for `bson_string_append_ex`) and assert as a documented precondition the given value is less than `UINT32_MAX`. This will address most of the new `-Wsign-conversion` and `-Wshorten-64-to-32` warnings in this PR.